src/cascadia/TerminalApp/Profile.cpp

```diff
@@ -794,6 +809,77 @@ std::string_view Profile::_SerializeCloseOnExitMode(const CloseOnExitMode closeO
     }
 }
 
+int32_t Profile::ParseSnapOnOutput(const Json::Value& json)
+{
+    const int32_t defaultValue = static_cast<int32_t>(SnapOnOutputFlag::AtBottom) | static_cast<int32_t>(SnapOnOutputFlag::NoSelection);
+
+    if (json.isString())
+    {
+        auto snapOnOutput = json.asString();
+        if (snapOnOutput == SnapOnOutputNever)
+        {
+            return static_cast<int32_t>(SnapOnOutputFlag::Never);
+        }
+        else if (snapOnOutput == SnapOnOutputNoSelection)
+        {
+            return static_cast<int32_t>(SnapOnOutputFlag::NoSelection);
+        }
+        else if (snapOnOutput == SnapOnOutputAtBottom)
+        {
+            return static_cast<int32_t>(SnapOnOutputFlag::AtBottom);
+        }
+        else if (snapOnOutput == SnapOnOutputAlways)
+        {
+            return static_cast<int32_t>(SnapOnOutputFlag::Always);
+        }
+    }
+    else if (json.isArray())
+    {
+        int32_t result = 0;
+        for (const auto value : json)
+        {
+            const auto snapOnOutput = value.asString();
+            if (snapOnOutput == SnapOnOutputNever || snapOnOutput == SnapOnOutputAlways)
+            {
+                // these values are not accepted
+                // this is considered a failure to parse and we fallback to the default value instead
+                return defaultValue;
+            }
+            else if (snapOnOutput == SnapOnOutputNoSelection)
+            {
+                result |= static_cast<int32_t>(SnapOnOutputFlag::NoSelection);
+            }
+            else if (snapOnOutput == SnapOnOutputAtBottom)
+            {
+                result |= static_cast<int32_t>(SnapOnOutputFlag::AtBottom);
+            }
+        }
+    }
+
+    return defaultValue;
+}
+
+std::string_view Profile::_SerializeSnapOnOutput(SnapOnOutputFlag snapOnOutput)
+{
+    if (snapOnOutput == SnapOnOutputFlag::Never)
+    {
+        return SnapOnOutputNever;
+    }
+    else if (snapOnOutput == SnapOnOutputFlag::Always)
+    {
+        return SnapOnOutputAlways;
+    }
+    else if (snapOnOutput == SnapOnOutputFlag::AtBottom)
+    {
+        return SnapOnOutputAtBottom;
+    }
+    else if (snapOnOutput == SnapOnOutputFlag::NoSelection)
+    {
+        return SnapOnOutputNoSelection;
+    }
+    return "";

```

Comment:
We don't actually use the "serialize" functions anywhere (I think). So, I think this is incorrect, but I also don't know what the "correct" behavior should be. We're also not using it so I guess I'll just worry about this when we start serializing again?